### PR TITLE
Feature/latest message bubble

### PR DESCRIPTION
**Pull request by AlexTeusz**
This feature request added the following features to the webchat:

1. Show badge with number of unseen bot messages
2. Show the lastest unseen bot message as a bubble next to the toggle button
3. Play a sound on each received message if the webchat is either closed or the tab is blurred

**Example:**

### CODE REVIEW

**Review by mastasky — APPROVED** [review]
Functional Review OK

**Comment by pedily** [comment]
This PR adds the concept of unssen messages
a message is unseen if a message arrives when either the webchat is closed or the tab is in the background
we added some features that can help users be aware of unread messages
- we added a badge that shows the number of unread messages at the toggle button
- we added a preview bubble that shows the text of the last message at the toggle button
- we added a notification sound that plays when unread messages arrive
- we added a title indicator that can be used to change the title of the page with an unread counter + a text

**Comment by kwinto** [comment]
If an unread message contains no extractable text, it should not hide the previous message preview.

An example:
- bot sends text message: "Just look at this cutie";
- bot sends a picture of a kitten;
It make sense to keep the first message preview instead of showing nothing as we currently do.

**Comment by kwinto** [comment]
Flashing page title indicator it pretty invasive. It would be nice to have an option to add unread message counter to the beginning of current page title without replacing/flashing it.

File: src/webchat-ui/components/presentational/Badge.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    const { content, backgroundColor, fontColor } = props;
+
+    const size = 24;
+
+    if (content === 0)
+        return null;
+
+    return (
+            <span style={{
```
Comment (kwinto) [inline src/webchat-ui/components/presentational/Badge.tsx:19]: Inline styles make it hard to customise. Also should provide a stable class name string for customisations.

**Comment by kwinto** [comment]
Sound notification does not work in Safari Desktop

File: docs/embedding.md [hunk 2]
```diff
@@ -68,6 +68,10 @@ See it in action:
 | enableSTT | boolean | false | x | x | Whether to load the speech input plugin.
 | enableTTS | boolean | false | x | x | Whether to load the speech output plugin. 
 | enableTypingIndicator | boolean | true | x | | Whether to enable typing indicators in the Webchat when the Conversational AI is replying. Requires a messageDelay to be set. 
+| enableUnreadMessageBadge | boolean | false | | | If true, the webchat shows a badge with the number of unread messages at the toggle button
+| enableUnreadMessagePreview | boolean | false | | | If true, the webchat shows a message bubble with the latest retreived bot message.
```
Comment (kwinto) [inline docs/embedding.md:72]: >retreived

**Comment by kwinto** [comment]
code looks good, please see the comment above

File: docs/embedding.md (hunk 2 above)
Comment (mastasky) [reply docs/embedding.md]: fixed

**Review by mastasky — CHANGES_REQUESTED** [review]
Message persistent can throw these features off.

1. Reload Issue
Get the webchat to display a preview and badge in closed mode.
Reload the page.
Badge is gone.
Click button.
Windows flashes and badge is back (wrongly)

2. Tab Change
Click something in the chat that will show messages
Quickly move to another tab
Messages come in and you see the title bar notification
Close the tab
Re-open a new tab with webchat url
Webchat is closed, open it. You will now hear sounds and badge shows
You must go to another tab and return to fix it

**Comment by pedily** [comment]
@mastasky issue 1 is fixed. the "open: false" state was restored on connect (the "open" event caused the connect!). The webchat will now stay open if it was open before

**Comment by pedily** [comment]
the other issue is also fixed

**Comment by mastasky** [comment]
Please make sure it say this:

(1) New Message
(2) New Messages

**Comment by mastasky** [comment]
> Please make sure it say this:
> 
> (1) New Message
> (2) New Messages

Built in

**Review by mastasky — APPROVED** [review]
lgtm

**Review by pedily — APPROVED** [review]

**Pull request merged**